app/src/main/java/com/dimelo/sampleapp/RcConfig.java

```diff
@@ -2,25 +2,32 @@
 
 import android.content.Context;
 import android.content.SharedPreferences;
+import android.util.Log;
 
 import com.dimelo.dimelosdk.main.Dimelo;
-import com.google.firebase.iid.FirebaseInstanceId;
 import com.huawei.hms.api.HuaweiApiAvailability;
 
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import java.io.InputStream;
+
 public class RcConfig {
       static final String RC_USER_ID = "rc_user_id";
       static final String RC_THREAD_ENABLED = "rc_thread_enabled";
+      static final String RC_SOURCE_NAME = "rc_source_name";
 
      static Dimelo setupDimelo(Context context) {
-        String secret = BuildConfig.ENGAGE_DIGITAL_MESSAGING_SECRET; // Edit in gradle.properties
-        String domainName = BuildConfig.ENGAGE_DIGITAL_DOMAIN_NAME; // Edit in gradle.properties
+        RcSourceModel rcSource = new RcSourceModel().getSelectedObject(context);
         Dimelo.setup(context);
-
         Dimelo dimelo = Dimelo.getInstance();
-        dimelo.initWithApiSecret(secret, domainName, null);
+
+        if ((rcSource.hostname != null) && !(rcSource.hostname.isEmpty())) {
+            dimelo.initializeWithApiSecretAndHostName(rcSource.apiSecret, rcSource.domainName + rcSource.hostname, null);
```

Comment:
`rcSource.domainName + rcSource.hostname` means that the hostname should contain the leading `.` otherwise it won't work

+ if someone just doesn't change the value and let `CHANGE_HERE` it won't work either we should be really clear about that in the readme

Reply:
it will raise an internal sdk exception with the right message and I think this is the same thing as above